backend/src/sms/middlewares/sms-template.middleware.ts

```diff
@@ -63,6 +64,40 @@ const storeTemplate = async (req: Request, res: Response, next: NextFunction): P
     return next(err)
   }
 }
+
+/**
+ * Updates the campaign and email_messages table in a transaction, rolling back when either fails.
+ * For campaign table, the s3 meta data is updated with the uploaded file, and its validity is set to true.
+ * For email_messages table, existing records are deleted and new ones are bulk inserted.
+ * @param key 
+ * @param campaignId
+ * @param filename 
+ * @param records
+ */
+const updateCampaignAndMessages = async (
```

Comment:
I am not sure if this function belongs here; considered putting it in one of the service files but doesn't really fit.